### ISSUE TITLE
Sign Messages with MicroSD

### ISSUE THREAD

**Issue by mjg-foundation** [sparrowwallet/sparrow#1491]
Would it be possible to add an option to sign messages with a given address via MicroSD from this screen? I'm working on the signing end of this with Passport, since some messages can be a bit too much to scan in a single QR code.
![sign_message](https://github.com/user-attachments/assets/9300c96f-9858-4be4-9836-5f9ce0cbf721)

**Comment by craigraw** [comment]
Do you mean saving a file in the Coldcard format? https://coldcard.com/docs/sign-text-file/#text-file-format

**Comment by mjg-foundation** [comment]
I'd like the standard (electrum) format if possible

**Comment by craigraw** [comment]
> I'd like the standard (electrum) format if possible

Is this documented anywhere?

**Comment by mjg-foundation** [comment]
Not anywhere great, but it's already exported via QR here in the sparrow ui, and it follows this format:
```
signmessage <derivation path> ascii: <message>
```
for example:
```
signmessage m/84h/0h/0h/0/0 ascii: chancellor on the brink of second bailout
```

**Comment by mjg-foundation** [comment]
I'll have to double check in the morning if my code strips leading spaces on the message. It wasn't too clear to me if a space would be inserted between the colon and the message as part of the standard.

**Comment by craigraw** [comment]
I think it would be better to use the same format as Coldcard, rather than introduce an effectively new format for this media? The format used for QRs is not really material, and is poorly documented as you say.

**Comment by mjg-foundation** [comment]
Ok, I can adapt to that. Thanks!

**Comment by craigraw** [comment]
Implemented in 17093dbf. As per the Coldcard spec the message is saved as a single line (newlines are removed) but no further checks or amendments are made.

<img width="568" alt="Screenshot 2024-09-03 at 12 06 10" src="https://github.com/user-attachments/assets/d285fa80-0c89-4b84-8808-bdb18cf65cf0">
